### DISCUSSION TITLE
2.x ER diagram event to listen click on items inside an entity

### DISCUSSION THREAD

**Discussion by yonas-adane** [antvis/X6#3098] [Q&A] [answered]
Looking at the ER demo for 2.x, is there an event which get fired when one of the items (port) inside an entity is clicked?

Here is the link: https://x6.antv.antgroup.com/examples/showcase/practices/#er

![98FAFC22-5A60-435F-B0F4-9F4653760CAF](https://user-images.githubusercontent.com/9521453/209889990-3a00938e-9e11-44e0-ba5b-b4ffa1fd673d.jpeg)

The resources i found online refer to events to nodes and edges - but not on ports or items inside an entity node.

**Comment by yonas-adane** [comment, marked as answer]
Finally figured this out - the example for [ToolTip](https://x6.antv.antgroup.com/examples/node/tool/#port-tooltip) was very helpful.

use the event "onPortRendered" to register an event listener. Then use the event to show data/attribute info inside a port. Here is an extract from the ToolTip example.

   ```
 const graph = new Graph({
      container: this.container,
      width: 800,
      height: 600,
      grid: true,
      onPortRendered({ contentContainer, port, node }) {
        const text = node.portProp(port.id, 'tip') as string
        registerPortTooltip(contentContainer, text) 
      },
    }) 

const registerPortTooltip = (container: Element, text: string) => {
  container.addEventListener('mouseenter', (e: MouseEvent) => {
    const tooltip = document.querySelector('.x6-tooltip') as HTMLElement
    const content = tooltip?.querySelector('.ant-tooltip-inner') as HTMLElement
    if (content) {
      content.innerText = text
      tooltip.style.left = `${e.clientX - content.offsetWidth / 2}px`
      tooltip.style.top = `${e.clientY - 50}px`
    }
  })
  container.addEventListener('mouseleave', () => {
    const tooltip = document.querySelector('.x6-tooltip') as HTMLElement
    tooltip.style.left = '-1000px'
    tooltip.style.top = '-1000px'
  })
}
```

**Reply by NewByVector** [reply]
Port related events have been added in the latest version of 2.x. Documentation is available [here](http://x6.antv.antgroup.com/tutorial/basic/events#%E9%BC%A0%E6%A0%87%E4%BA%8B%E4%BB%B6).
